Sources/Network/JSON.swift

```diff
@@ -186,6 +186,61 @@ public enum JSON {
         return value
     }
 
+    /// Parse an attribute of type `T: Mappable` with the given key on the given JSON dictionary, validating if
+    /// the value can be used to instantiate a Mappable model. Additionally, an optional `parseAPIError` closure can be supplied
+    /// so that if a parsing step fails an attempt is made to extract a domain specific error and throw it.
+    ///
+    /// - Parameters:
+    ///   - type: The type of the attribute to parse.
+    ///   - key: The JSON attribute key to parse.
+    ///   - json: The JSON dictionary.
+    ///   - parseAPIError: The API error parsing closure.
+    /// - Returns: The value of type `T` associated to the given attribute's key.
+    /// - Throws: An error of type `JSON.Error`, or a domain specific `Swift.Error` produced by `parseAPIError`.
+    public static func parseMappableAttribute<T: Mappable>(
+        _ type: T.Type,
+        key: JSON.AttributeKey,
+        json: JSON.Dictionary,
+        parseAPIError: ParseAPIErrorClosure? = nil) throws -> T {
+
+        guard let anyValue = json[key] else {
+            throw parseAPIError?(json) ?? Error.missingAttribute(key, json: json)
+        }
+
+        let anyObject: Any = try parseValue(rawValue: anyValue, key: key, json: json, where: nil, parseAPIError: parseAPIError)
```

Comment:
since this is an `Any`, perhaps call this `let any = ...` ? 💄 
(same applies for the other places below)